### ISSUE TITLE
Type Error on locale when using Next provided type GetStaticProps

### ISSUE THREAD

**Issue by yoont4** [isaachinman/next-i18next#1307]
## Describe the bug
using the default NextJS typings for `getStaticProps`, locale is potentially undefined, which throws a type error:

```
Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2345)
```

## Occurs in next-i18next version
`8.4.0`

## Steps to reproduce
Use this code for `getStaticProps`

``` typescript
export const getStaticProps: GetStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale, ['common'], i18nConfig)),
  },
})
```

## Expected behaviour
It shouldn't throw a type error

## Screenshots
![image](https://user-images.githubusercontent.com/7411742/124826867-32de9300-df2a-11eb-822e-8427dcc7bd0a.png)

## OS (please complete the following information)
 - Device: Windows Desktop
 - Browser: Chrome

## Additional context
Not sure if `locale` is actually ever undefined as long as a `defaultLocale` is defined in `next.config.js`, but I don't like having to override it since it throws an eslint `no-non-null-assertion` warning.

i.e. ` serverSideTranslations(locale!, ['common'], i18nConfig))`

**Comment by isaachinman** [comment]
What do you propose? The typings from NextJs _are_ correct – locale can be `undefined` if i18n isn't set up. You'll need to cast the param in your codebase.

**Comment by yoont4** [comment]
That's fair

**Comment by yfhui** [comment]
@isaachinman Hi I am using next-i18next and I found that locale is `undefined` when hitting a 500 page using `getStaticProps` (in dev enviornment). What is the possible explanation of this?

Some directions I have explored:
* The doc mentioned that 500 page could use `getStaticProps`, so locale should be defined. (i18n is functioning at other pages and the error is introduced intentionally to trigger test the 500 page)
* I could also see that [#18513](https://github.com/vercel/next.js/pull/18513) addresses a similiar issue at 404 page. Is there any relationship to that pull request?
* In production, getStaticProps will build the 500 page for all locales, so locale is defined and `serverSideTranslations` works perfectly
* I could see that 500 page `getStaticProps` run twice, 1st time with the locale defined, but 2nd time with locale `undefined`, I could not see why page 500 could run twice (I could access it directly using `localhost:3000/500` so page 500 doesn't have any error)

Many thanks!

**Comment by isaachinman** [comment]
@yfhui The short answer is that `locale` is handled by NextJs directly and is not something that `next-i18next` has any control over. It was not until quite recently that the 404 page was possible to localise at all, so I suspect your 500 page faces similar issues.

Can you please search for, or create a new issue, on the NextJs repo and link it here?

**Comment by yfhui** [comment]
@isaachinman please check the main issue, I have added more comments, thanks for your quick reply!
